hpx/lcos/server/barrier.hpp

```diff
@@ -82,6 +82,8 @@ namespace hpx { namespace lcos { namespace server
             }
         }
 
+        HPX_DEFINE_COMPONENT_ACTION(barrier, set_event, set_event_non_direct_action);
+
```

Comment:
We might want to move this new action to the `base_lco` base class as other LCO's might require this functionality as well.